summer-23-unity/Assets/Scripts/PlayerHealth.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class PlayerHealth : MonoBehaviour
+{
+    public int maxHealth = 100;
+    public int currentHealth;
+    private bool heldFlag = true;
+    // Start is called before the first frame update
+    void Start()
+    {
+        currentHealth = maxHealth;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if(Input.GetKeyDown(KeyCode.R)){
+            if(!heldFlag){
+                Kill();
+            }
+        }
+        else{
+            heldFlag=false;
+        }
+        
+        
+    }
+
+    public void Damage(int amount){
+        if(currentHealth-amount<=0){
+            Kill();
+        }
+        else{
+            currentHealth-=amount;
+        }
+    }
+
+    public void Damage(float amount){
+        Damage((int)amount);
+    }
+
+
+    public void Kill(){
+        currentHealth = maxHealth;
+        Debug.Log("Died");
+        Scene scene = SceneManager.GetActiveScene(); SceneManager.LoadScene(scene.name);
```

Comment:
Are you running a linter on your code before committing? it should have caught this case of 2 lines of code on 1 line in the file.